### ISSUE TITLE
Minimum Node Version Fails to Run Firebase CLI tools

### ISSUE THREAD

**Issue by zzeleznick** [actions-on-google/dialogflow-facts-about-google-nodejs#17]
## Description
Currently, with the suggested node version [`~4.2`](https://github.com/actions-on-google/apiai-facts-about-google-nodejs/blob/master/functions/package.json#L9) for this project and the latest [Firebase CLI tools](https://firebase.google.com/docs/functions/get-started#set_up_and_initialize_functions_sdk) version [`3.12.0`](https://github.com/firebase/firebase-tools/commit/faa7a77140c3438bb18fa58e506930269ceed4b6), running `firebase init` will fail:

```
$ firebase init
/usr/local/lib/node_modules/firebase-tools/node_modules/tar/lib/pack.js:32
const EOF = Buffer.alloc(1024)
                   ^

TypeError: Buffer.alloc is not a function
```
It seems that can be fixed by bumping up the minimum node version, as `Buffer.alloc` was added in Node [`4.5.0`](https://github.com/lovell/sharp/issues/847)

## Proposal
1. ~~Update the node version to minimum requirement, `^4.5`, as [Firebase tools](https://github.com/firebase/firebase-tools/blob/master/package.json#L29) does.~~ See Update below
1. Or, use the node version recommended by [Firebase CLI docs](https://firebase.google.com/docs/functions/get-started#set_up_and_initialize_functions_sdk) `v.6.11.1`

**Comment by zzeleznick** [comment]
### Update
Running `firebase serve --only functions` with node `v4.6.2` and firebase-tools version `3.12.0`, I see the following error:

```sh
⚠  functions: Cannot start emulator. Please use Node version v6.9.1, you have v4.6.2
```
With node `v6.9.1` and firebase-tools version `3.12.0`, I see the following error:

```
i  functions: Preparing to emulate functions.
Node.js v6.11.x or greater is required to run the Emulator!
⚠  functions: Failed to load functions source code. Ensure that you have the latest SDK by running npm i --save firebase-functions inside the functions directory.
⚠  functions: Error from emulator. Error: Emulator crashed! Check the log file...
```

node `^6.11` should likely be the specified minimum version if using the latest firebase-tools is still recommended.

I will update my proposal above to use a later version.

**Comment by zzeleznick** [comment]
@smishra2: Thoughts?

**Comment by Canain** [comment]
Fixed now with 1293c17ca936f2e2b85e40492d69c5bcf445b192 in [package.json](https://github.com/actions-on-google/dialogflow-facts-about-google-nodejs/blob/master/functions/package.json#L9)
